cmd/az-analyze/cmd/azv_test.go

```diff
@@ -0,0 +1,194 @@
+                        Name: "volume-name-0",
+                        VolumeSource: v1.VolumeSource {
+                            PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource {
+                                ClaimName: "test-pvcClaimName",
+                            },
+                        },
+                    },
+                    {
+                        Name: "volume-name-1",
+                        VolumeSource: v1.VolumeSource {
+                            PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource {
+                                ClaimName: "test-pvcClaimName-1",
+                            },
+                        },
+                    },
+                },
+            },
+        }, &v1.Pod {
+            ObjectMeta: metav1.ObjectMeta {
+                Name: "test-pod-1",
+                Namespace: "default",
+            },
+            Spec: v1.PodSpec {
+                Volumes: []v1.Volume {
+                    {
+                        Name: "volume-name",
+                        VolumeSource: v1.VolumeSource {
+                            PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource {
+                                ClaimName: "test-pvcClaimName",
+                            },
+                        },
+                    },
+                },
+            },
+    })
+}
+
+func NewTestAzDiskClientset() *diskfakes.Clientset {
+    return diskfakes.NewSimpleClientset(&v1beta1.AzVolume {
+            ObjectMeta: metav1.ObjectMeta {
+                Name: "test-azVolume",
+                Namespace: "azure-disk-csi",
+            },
+            Spec: v1beta1.AzVolumeSpec{
+                VolumeName: "test-azVolume",
+                Parameters: map[string]string{consts.PvcNameKey: "test-pvcClaimName"},
+            },
+            Status: v1beta1.AzVolumeStatus{
+                State: v1beta1.VolumeCreated,
+            },
+        }, &v1beta1.AzVolume {
+            ObjectMeta: metav1.ObjectMeta {
+                Name: "test-azVolume-1",
+                Namespace: "azure-disk-csi",
+            },
+            Spec: v1beta1.AzVolumeSpec{
+                VolumeName: "test-azVolume-1",
+                Parameters: map[string]string{consts.PvcNameKey: "test-pvcClaimName-1"},
+            },
+            Status: v1beta1.AzVolumeStatus{
+                State: v1beta1.VolumeCreated,
+            },
+    })
+}
+
+func TestGetAzVolumesByPod(t *testing.T) {
+    fakeClientsetK8s := NewTestK8sClientset()
+    fakeClientsetAzDisk := NewTestAzDiskClientset()
+
+    tests := []struct {
+        description string
+        verifyFunc func()
+    }{
+        {
+            description: "specified pod name with more than one pv",
+            verifyFunc: func() {
+                result := GetAzVolumesByPod(fakeClientsetK8s, fakeClientsetAzDisk, "test-pod-0", "default")
+                expect := []AzvResource {
+                    {
```

Comment:
We can create AzvResource outside of each test case, so that we can reuse them for all test cases.

e.g. azvResource1 :=  {
                        ResourceType: "test-pod-0",
                        Namespace:    consts.DefaultAzureDiskCrdNamespace,
                        Name:         "test-azVolume",
                        State:        v1beta1.VolumeCreated,
                    },

 azvResource2 := azvResource1 :=  {
                        ResourceType: "test-pod-0",
                        Namespace:    consts.DefaultAzureDiskCrdNamespace,
                        Name:         "test-azVolume-1",
                        State:        v1beta1.VolumeCreated,
                    },
 And then under expect for each test case, we can write []AzvResource{azvResource1 , azvResource2}